### PR TITLE
Fix sync multiple

### PR DESCRIPTION
**Pull request by den-sergeyev**
Wrap Savon response to return always array.

Jira issue:
https://jira.backupify.org/browse/BETA-570

### CODE REVIEW

File: lib/marketo/interface.rb [hunk 1]
```diff
@@ -68,7 +68,14 @@ def sync_multiple(leads_array)
...
 
       response = send_request("ns1:paramsSyncMultipleLeads", lead_record_list: { lead_record: lead_records })
-      response[:success_sync_multiple_leads][:result][:sync_status_list][:sync_status]
+      wrap_response(response[:success_sync_multiple_leads][:result][:sync_status_list][:sync_status])
+    end
+
+    private
+
+    # Wraps single lead response into array
+    def wrap_response(response)
```
Comment (Alexey-Tarasevich-epam) [inline lib/marketo/interface.rb]: i would call it `normalize_response` or `normalize_sync_status`

File: spec/interface_spec.rb [hunk 2]
```diff
@@ -2,18 +2,36 @@
 
 describe Marketo do
   describe Marketo::Interface do
+    before(:each) do
+      @client = Marketo::Client.new_marketo_client
```
Comment (Alexey-Tarasevich-epam) [inline spec/interface_spec.rb]: use `let(:client) { Marketo::Client.new_marketo_client }`

File: spec/interface_spec.rb [hunk 3]
```diff
@@ -2,18 +2,36 @@
...
 
         first_timestamp.should_not == second_timestamp
 
         Timecop.return
       end
     end
+
+    describe 'wrap_response' do
+      it 'should return array if response is array' do
+        @client = Marketo::Client.new_marketo_client
```
Comment (Alexey-Tarasevich-epam) [inline spec/interface_spec.rb]: I think you don't needed it here - you define it in before.

File: spec/interface_spec.rb [hunk 4]
```diff
@@ -2,18 +2,36 @@
...
+        response = [{:lead => 12345, :status => 'UPDATED', :error => nil},
+                    {:lead => 12346, :status => 'SKIPPED', :error => 'Not uniq'}]
+
+        response.should == @client.send(:wrap_response, response)
```
Comment (Alexey-Tarasevich-epam) [inline spec/interface_spec.rb]:
```
@client.send(:wrap_response, response).should == expected_response
```

opposite order comparing to XUnit libraries.

File: spec/interface_spec.rb [hunk 5]
```diff
@@ -2,18 +2,36 @@
...
         first_timestamp.should_not == second_timestamp
 
         Timecop.return
       end
     end
+
+    describe 'wrap_response' do
+      it 'should return array if response is array' do
+        @client = Marketo::Client.new_marketo_client
+        response = [{:lead => 12345, :status => 'UPDATED', :error => nil},
```
Comment (Alexey-Tarasevich-epam) [inline spec/interface_spec.rb:171]: consider renaming to `expected_response`

File: spec/interface_spec.rb [hunk 6]
```diff
@@ -2,18 +2,36 @@
...
+                    {:lead => 12346, :status => 'SKIPPED', :error => 'Not uniq'}]
+
+        response.should == @client.send(:wrap_response, response)
+      end
+
+      it 'should return one element array if response is hash' do
+        response = {:lead => 12345, :status => 'UPDATED', :error => nil}
+        [response].should == @client.send(:wrap_response, response)
```
Comment (Alexey-Tarasevich-epam) [inline spec/interface_spec.rb]: same

File: spec/marketo_spec.rb [hunk 7]
```diff
@@ -120,6 +120,19 @@
         response.should == test_values
       end
 
+      it "should sync single lead and return array as result" do
+        multi_users = [
+          { "Email" => "admin@backupify.org", "FirstName" => "Reed", "LastName" => "Richards" }
+        ]
+        test_values = [
```
Comment (Alexey-Tarasevich-epam) [inline spec/marketo_spec.rb]: `expected_result` ? why "test_values" ?

File: spec/interface_spec.rb (hunk 5 above)
Comment (den-sergeyev) [inline spec/interface_spec.rb:171]: I believe that response here is actual parameter that we pass to testing method. That's why I believe response is descriptive enough.

**Comment by Alexey-Tarasevich-epam** [comment]
LGTM

**Comment by ericalexander** [comment]
LGTM

**Pull request merged**